### PR TITLE
Main helm chart deployment

### PR DESCRIPTION
**Pull request by huseyinbabal**
## Description

Changes proposed in this pull request:

- It generates additional docker images with version containing commit id
- Publishes the chart into gcs

## Proof

You can see some testings I did on my personal git account below

Screenshot 2023-10-04 at 11 22 26

### Publishing Helm Chart
https://github.com/huseyinbabal/botkube/actions/runs/6391937992/job/17348177345

### Helm Chart Repo Add & Listing Versions
```bash
helm repo add gcs-botkube https://storage.googleapis.com/dev-botkube-latest-main-charts
helm repo update gcs-botkube
```

```bash
helm search repo gcs-botkube/botkube --devel
```

## Related issue(s)

Relates #1203

### CODE REVIEW

File: .github/workflows/branch-build.yml [hunk 1]
```diff
@@ -11,12 +11,29 @@ env:
   HELM_VERSION: v3.9.0
   K3D_VERSION: v5.4.6
   IMAGE_REGISTRY: "ghcr.io"
-  IMAGE_REPOSITORY: "kubeshop/botkube"
-  CFG_EXPORTER_IMAGE_REPOSITORY: "kubeshop/botkube-config-exporter"
+  IMAGE_REPOSITORY: "huseyinbabal/botkube"
```
Comment (huseyinbabal) [inline .github/workflows/branch-build.yml:14]: will be reverted before merge

File: .github/workflows/branch-build.yml [hunk 2]
```diff
@@ -38,16 +55,19 @@ jobs:
         with:
           registry: ghcr.io
           username: ${{ github.actor }}
-          password: ${{ secrets.GITHUB_TOKEN }}
+          password: ${{ secrets.GH_TOKEN }}
```
Comment (huseyinbabal) [inline .github/workflows/branch-build.yml:58]: will be reverted before main merge

File: .github/workflows/branch-build.yml (hunk 1 above)
Comment (huseyinbabal) [reply .github/workflows/branch-build.yml]: reverted

**Review by josefkarasek — APPROVED** [review]
lgtm

**Pull request merged**